examples/QIR/Simulation/TraceSimulator/QubitManager.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "CoreTypes.hpp"
+
+namespace Microsoft
+{
+namespace Quantum
+{
+    class QubitManager
+    {
+        // Keep track of unique qubit ids via simple counter.
+        uint64_t nextQubitId = 0;
+
+        // Get the internal ID associated to a qubit object.
+        static uint64_t GetQubitId(Qubit qubit)
+        {
+            return reinterpret_cast<uint64_t>(qubit);
+        }
+
+      public:
+        Qubit AllocateQubit()
+        {
+            return reinterpret_cast<Qubit>(this->nextQubitId++);
+        }
+
+        void ReleaseQubit(Qubit qubit)
+        {
```

Comment:
It might be worth noting here that not decrementing the `nextQubitId` means that no id will ever be reused. This may be intentional, but should probably be called out, as it differs from the behavior in the State Simulator.